app/assets/javascripts/surveys.js

```diff
@@ -194,15 +202,29 @@
       userSurveys.trackEvent(survey.identifier, 'banner_shown', 'Banner has been shown')
     },
 
+    initialiseModuleOnElement: function (element, Module) {
+      if (Module) {
+        var module = new Module(element)
+        module.init()
+        return module
+      }
+    },
+
     displayURLSurvey: function (survey, surveyContainer) {
       var urlSurveyTemplate = userSurveys.getUrlSurveyTemplate()
       surveyContainer.innerHTML = urlSurveyTemplate.render(survey)
+      this.initialiseModuleOnElement(surveyContainer, window.GOVUK.Modules.Ga4LinkTracker)
+      this.initialiseModuleOnElement(surveyContainer, window.GOVUK.Modules.Ga4EventTracker)
```

Comment:
What's the event tracker used for? I'm guessing something to do with the form, but I don't see a `data-ga4-event` anywhere.

Reply:
It's on line 26, for the `Close` button on the banner

Reply:
As it acts like a button as it closes the banner, we want to track it like a button - if we track it as a link, we would get values `url`, `link_path_parts`, `method`, and `link_domain` being sent to the dataLayer.

Reply:
Ah of course, thanks for the explanation.